### ISSUE TITLE
Determine distance between two points on selected track/route

### ISSUE THREAD

**Issue by smartyw** [organicmaps/organicmaps#5592]
**Is your feature request related to a problem? Please describe.**
I use Organic Maps when hiking and bikepacking. I plan my route using another tool, export a KML file and then open this route in Organic Maps where I believe it is called a Track.

I often need to know how far it is between two points, following my loaded track. For example, I may want to know the distance between my current position and a place on the route ahead of me. Currently I can use Route To, but this calculates its own route and reports the distance along that calculated route. That may be approximately correct if I'm lucky, but I'd like to know more precisely the distance and ascent on the route I am following, not the route that Organic Maps suggests I follow. 

**Describe the solution you would like**
I would like a new Routing Option "Use selected track for route calculations" or similar. This would be a switch.
I would then like to be able to select a point on the map and on selecting Route To, a route that goes from my current position to the selected destination point, using the best of one or more of the tracks that are active on screen for as much of the route as possible, would be calculated and the usual results and options displayed. 

**Describe alternatives you have considered**
- I look at the displayed scale and guestimate the distance. When the route is a winding road or river though, this is often very inaccurate.

**Additional context**

**Comment by biodranik** [comment]
Is it about checking a straight line distance between any points? That is a separate issue/feature.
To see a straight distance from your current position to any point on the map, select that point on the map, and see the distance (and direction arrow) on the bottom info panel (you can even click it).

If it is about displaying a distance for a track segment, then it's a different issue. Note that without refactoring of elevation data, it may not be possible to see elevation if there is no underlying OSM track data.

**Comment by smartyw** [comment]
It's about distance along a track segment. I added q screenshot showing two points on a KML route I am currently following. It's very much not straight and wanting to know distances between points on it, where typically one of the points is my current position is a very common use case which generally arises several times each day. 

![Uploading Screenshot_2023-07-24-06-27-07-71_75d0290a7c00525acd83c0ab8700d64c~2.jpg…]()

**Comment by biodranik** [comment]
The screenshot does not work. Maybe you can record a screen video of how it works in another app?

**Comment by smartyw** [comment]
Hi, sorry the screenshot doesn't help. And I'm not aware of another app that has this feature. Organic Maps and Google Maps are all I use. The concept is simple though, I think. Determine the distance and (if possible) ascent, from the current location to another selected location, using one or more of the tracks that are active for all or the major part of the route. 

I don't want to stipulate a design because the core team are better qualified to do this. But for example, as I suggested, introduce a new Routing Option, "Use selected tracks" or something. Then when you select Route To, you'd perhaps prompt the user with a list of active tracks to be used in the calculation and then, determine the shortest route from the current location to the nearest point on one of the selected tracks and calculate the distance. Them calculate the distance along this track to the destination, possibly involving other tracks if more than one was selected.

In short, we're answering the question "How much further do I need to walk / cycle on my route to get to this point?".

Hope that makes sense. I'm currently in a bikepacking trip so I'm a bit limited as to what I can do, here.

Thanks

**Comment by biodranik** [comment]
Looks like you need #1360, then you'll see the distance to the next stop on the track.

**Comment by smartyw** [comment]
Maybe, yes. Although I don't want navigation. I'm doing that manually by occasionally visually inspecting my position relative to the active track as this way uses as little battery as possible. But if #1360 will provide the functionality I described without the need to engage navigation then that sounds good. All the best and keep up the good work :-)
